test/tests/test_optional_argument.cpp

```diff
@@ -0,0 +1,387 @@
+#define AP_TESTING
+
+#include "../doctest.h"
+#include "../testing_friend_definitions.hpp"
+
+#include <ap/argument_parser.hpp>
+
+#include <string_view>
+
+using namespace ap::detail;
+
+
+namespace {
+
+using test_value_type = int;
+
+constexpr std::string_view long_name = "test";
+constexpr std::string_view short_name = "t";
+
+
+optional_argument<test_value_type> default_optional_argument_long_name() {
+    return optional_argument<test_value_type>(long_name);
+}
+
+optional_argument<test_value_type> default_optional_argument_both_names() {
+    return optional_argument<test_value_type>(long_name, short_name);
+}
+
+} // namespace
+
+
+
+TEST_CASE("optional argument with only long name should be optional and not "
```

Comment:
There is no need to create two test cases for each functionality if the test cases differ only in specifying the long or both argument names.